cpp/cmake/thirdparty/CUDF_GetThrust.cmake

```diff
@@ -0,0 +1,33 @@
+#=============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_thrust VERSION)
+    CPMFindPackage(NAME Thrust
+        VERSION         ${VERSION}
+        GIT_REPOSITORY  https://github.com/NVIDIA/thrust.git
+        GIT_TAG         ${VERSION}
+        GIT_SHALLOW     TRUE
+        # If there is no pre-installed thrust we can use, we'll install our fetched copy together with cuDF
+        OPTIONS         "THRUST_INSTALL TRUE"
```

Comment:
Do we need to install thrust with cudf? I believe we've made it a point to avoid leaking Thrust in libcudf headers.

Reply:
Good catch, I don't think we need to install it alongside cudf. I copied RMM's behavior, but forgot RMM is header-only and needs Thrust to work.